src/main/java/com/google/devtools/build/lib/runtime/CommandEnvironment.java

```diff
@@ -329,7 +330,7 @@ public void exit(AbruptExitException exception) {
         }
       }
       for (Map.Entry<String, String> entry :
-          options.getOptions(CoreOptions.class).testEnvironment) {
+          options.getOptions(TestOptions.class).testEnvironment) {
```

Comment:
@katre I'm not 100% sure whether this will work as is - would `build` and `info` need to declare that they use `TestOptions`?

Reply:
Just add a check that TestOptions is present: if it's absent the entire block can be skipped.

Reply:
I'm worried that it just won't ever be there for `info`. The tests didn't trigger any crashes though.

Reply:
if you wrap the entire for loop in `if (options.contains(TestOptions.class))` then it'll be safe for any command.

Reply:
It will be safe in the sense of "doesn't crash", but will this still be a backwards compatible change in the sense that `bazel info client-env` shows test env entries?

Reply:
Yes, I was able to run a quick demo:

```
$ FOO=bar ~/repos/bazel/bazel-bin/src/bazel --host_jvm_debug info --test_env=FOO client-env
Running host JVM under debugger (listening on TCP port 5005).
build --test_env=FOO=bar
```

Reply:
SGTM.